### DISCUSSION TITLE
On using GitHub + backups

### DISCUSSION THREAD

**Discussion by HeyMyian** [vkbo/novelWriter#1049] [General]
Yo. I'm fairly new to git and GitHub and was wondering about how novelWriter's backup feature plays together with that.

I noticed my backup zips jumped up in size as soon as I set the project up as a local repository. I'm using GitHub Desktop to mirror the local files to an online repository every day, in case that's relevant. novelWriter now also backups the hidden `.git` folder that resides within the project folder, which currently is about twice the size of the actual project. The current sizes aren't a concern yet because they are so small, I'm talking ca. 300KB and 800KB respectively, creating a zip of about 500KB in total.

But I'm wondering if this is intended behavior and what any benefits of this are, to have novelWriter include everything within the project folder and not just the files novelWriter expects to be in there. Do some people perhaps stash other data (character images and whatnot) within the project folder that would make this necessary? Because I'm not and I'm heavily disinclined to do that anyway.

I'm probably misunderstanding something about the functionality here, so I'm seeking clarification or your advice on how to do it properly, in case I'm not yet.

**Comment by vkbo** [comment]
As you've noticed, the backup feature just zips the project folder. As a rule, you shouldn't be adding any files inside that folder other than what novelWriter adds. It is much safer to just leave it alone. If you need to store other files with the project, you should do that one level up. Ideally, that also includes the git data. If you do choose to save files in that folder anyway, then they will be included in the zip.

If you already push your changes to GitHub, then there may not be much need for the zipped backups anyway, so you could just turn it off for the relevant project. You can override the default settings in Project Settings. The built-in backup feature is meant to be a minimal implementation for people who have no general backups set up.
